modules/installation-user-infra-machines-advanced-customizing-live-iscsi-ibft.adoc

```diff
@@ -0,0 +1,65 @@
+
+ifeval::["{boot-media}" == "ISO image"]
+. Retrieve the {op-system} ISO image from the link:https://mirror.openshift.com/pub/openshift-v4/dependencies/rhcos/latest/[{op-system} image mirror] page and run the following command to customize the ISO image with the following information:
++
+[source,text]
+----
+$ coreos-installer iso customize \
+    --pre-install mount-iscsi.sh \ <1>
+    --post-install unmount-iscsi.sh \ <2>
+    --dest-device /dev/mapper/mpatha \ <3>
+    --dest-ignition config.ign \ <4>
+    --dest-karg-append rd.iscsi.firmware=1 \ <5>
+    --dest-karg-append rd.multipath=default \ <6>
+    -o custom.iso rhcos-<version>-live.x86_64.iso
+----
+<1> The script that gets run before installation. It should contain the `iscsiadm` commands for mounting the iSCSI target and any commands enabling multipathing.
+<2> The script that gets run after installation. It should contain the command `iscsiadm --mode node --logout=all`.
+<3> The path of a single multipathed device. If there are multiple multipath devices connected, or to be explicit, you can use the World Wide Name (WWN) symlink available in `/dev/disk/by-path`.
+<4> The ignition configuration for the destination system.
+<5> The iSCSI parameter is read from the BIOS firmware.
+<6> Optional: include this parameter if you are enabling multipathing.
+endif::[]
+
+ifeval::["{boot-media}" == "PXE environment"]
+. Retrieve the {op-system} `kernel`, `initramfs` and `rootfs` files from the link:https://mirror.openshift.com/pub/openshift-v4/dependencies/rhcos/latest/[{op-system} image mirror] page and run the following command to create a new customized `initramfs` file with the following information:
++
+[source,text]
+----
+$ coreos-installer pxe customize \
+    --pre-install mount-iscsi.sh \ <1>
+    --post-install unmount-iscsi.sh \ <2>
+    --dest-device /dev/mapper/mpatha \ <3>
+    --dest-ignition config.ign \ <4>
+    --dest-karg-append rd.iscsi.firmware=1 \ <5>
+    --dest-karg-append rd.multipath=default \ <6>
+    -o custom.img rhcos-<version>-live-initramfs.x86_64.img
+----
+<1> The script that gets run before installation. It should contain the `iscsiadm` commands for mounting the iSCSI target.
+<2> The script that gets run after installation. It should contain the command `iscsiadm --mode node --logout=all`.
+<3> The path of a single multipathed device. If there are multiple multipath devices connected, or to be explicit, you can use the World Wide Name (WWN) symlink available in `/dev/disk/by-path`.
+<4> The ignition configuration for the destination system.
+<5> The iSCSI parameter is read from the BIOS firmware.
+<6> Optional: include this parameter if you are enabling multipathing.
+endif::[]
++
+For more information about the iSCSI options supported by `dracut`, see the link:https://www.man7.org/linux/man-pages/man7/dracut.cmdline.7.html[`dracut.cmdline` manual page].
```

Comment:
I love how you set this up. It's clean and easy to read. But I have a question: are the scripts you are talking about provided by us at Red Hat or do they need to be written by the user? (and, please pardon me if I missed something along the way and should have known this)

Reply:
The `dracut` commands i believe are inherited from RHEL

Reply:
@GroceryBoyJr ohh i see what you mean. you're referring to the `pre-install` and `post-install` scripts mentioned. they need to be created by the user so kind of outside the scope of what we document. however, they need to include the specific things mentioned + whatever else they need for their infrastructure